src/txmempool.h

```diff
@@ -816,6 +790,102 @@ class CTxMemPool
         assert(m_epoch.guarded()); // verify guard even when it==nullopt
         return !it || visited(*it);
     }
+
+    /*
+     * CTxMemPoolChangeSet:
+     *
+     * This class is used for all mempool additions and associated removals (eg due to rbf).
+     * Callers can invoke StageAddition()/StageRemoval() of transactions in any order.
+     *
+     * CalculateChunksForRBF() can be used to calculate the feerate diagram of the proposed
+     * set of new transactions and comare with the existing mempool.
+     *
+     * CalculateMemPoolAncestors() calculates the in-mempool (not including
+     * what is in the change set itself) ancestors of a given transacion.
+     *
+     * Apply() will apply the removals and additions that are staged into the* mempool.
+     *
+     * Only one changeset may exist at a time. While a changeset is outstanding, no removals
```

Comment:
when would removals or additions happen outside of a changeset?

Reply:
Additions never happen outside a changeset.  RBF-related removals always go through a changeset as well.

Removals that happen because a block is found, or due to a reorg, or due to mempool expiry or mempool limiting do not go through the changeset.